test/rx_ruby/core/test_observable_creation.rb

```diff
@@ -463,21 +462,179 @@ def range_dispose
   end
 
   # Repeat methods
-
 =begin
+# the clock is actually off, because of not using the `scheduler.schedule_recursive`
   def test_repeat_value_count_zero
     scheduler = RxRuby::TestScheduler.new
 
-    res = scheduler.configure(:dispose => 204) do
+    res = scheduler.configure do
       RxRuby::Observable.repeat(42, 0, scheduler)
     end
 
     msgs = [
       on_completed(201)
     ]
-    assert_messages msgs, res.messages 
+    assert_messages msgs, res.messages
+  end
+
+  def test_repeat_value_once
+    scheduler = RxRuby::TestScheduler.new
+
+    res = scheduler.configure do
+      RxRuby::Observable.repeat(42, 1, scheduler)
+    end
+
+    msgs = [
+        on_next(201, 42),
+        on_completed(202)
+    ]
+    assert_messages msgs, res.messages
   end
 =end
 
+  def test_repeat_infinitely_dispose
```

Comment:
@minamimonji The previous code was good according to the Ruby [spec](https://github.com/ruby/spec/blob/master/language/return_spec.rb). It looks like, the Rubinius is a little bit special :) This change should not change the semantics of the code, though.